StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift

```diff
@@ -139,11 +154,31 @@ import UIKit
         }
 
         self.selection = selection
+        updateMandate()
     }
 
     func didTapAccessoryButton() {
         // TODO(porter)
     }
+
+    // MARK: Mandate handling
+    private func updateMandate(updateHeight: Bool = true) {
+        let previousHeight = frame.size.height
+        self.mandateView.attributedText = mandateProvider.mandate(for: selection?.paymentMethodType, savedPaymentMethod: selection?.savedPaymentMethod, bottomNoticeAttributedString: nil)
+
+        self.mandateView.setHiddenIfNecessary(self.mandateView.attributedText?.string.isEmpty ?? true)
+
+        guard updateHeight else { return }
+
+        // Force the layout to update so we get the new height
+        setNeedsLayout()
+        layoutIfNeeded()
+
+        let newHeight = frame.size.height
+        if previousHeight != newHeight {
+            delegate?.heightDidChange()
+        }
```

Comment:
I wonder if there's a way we could do this more generically. Like override `layoutSubviews` and check there.

Reply:
Ah good idea, I overrode `layoutSubviews`, only caveat is I needed an instance variable to keep track of the previous height.